### PR TITLE
rootless: set sticky bit on rundir

### PR DESCRIPTION
**Pull request by giuseppe**
it prevents the directory to be auto pruned, according to the XDG specifications.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: libpod/runtime.go [hunk 1]
```diff
@@ -309,7 +309,17 @@ func getDefaultTmpDir() (string, error) {
...
 		return "", err
 	}
-	return filepath.Join(rootlessRuntimeDir, "libpod", "tmp"), nil
+	libpodRuntimeDir := filepath.Join(rootlessRuntimeDir, "libpod")
+
+	if err := os.Mkdir(libpodRuntimeDir, 0700|os.ModeSticky); err != nil {
+		if !os.IsExist(err) {
+			return "", errors.Wrapf(err, "cannot mkdir %s", libpodRuntimeDir)
+		} else if err := os.Chmod(libpodRuntimeDir, 0700|os.ModeSticky); err != nil {
+			// The directory already exist, just set the sticky bit
```
Comment (mheon) [inline libpod/runtime.go]: Can you move this comment up a line? My brain parses it as being attached to the `errors.Wrapf()`

**Comment by mheon** [comment]
One small comment otherwise LGTM

**Comment by mheon** [comment]
Alright, we want to cut a release, it's not worth cutting a new version when this passes tests...
/lgtm

**Pull request merged**